scenario/3_struct_identity_mapping_pointer_error_wrapping.yml

```diff
@@ -0,0 +1,89 @@
+input:
+    input.go: |
+        package structs
+
+        // goverter:converter
+        // goverter:extend strconv:Atoi
+        type Converter interface {
+            Convert(source []Person) ([]APIPerson, error)
+
+            // goverter:mapIdentity Address
+            ConvertPerson(source Person) (APIPerson, error)
+
+            ConvertAddress(source Person) (APIAddress, error)
+        }
+
+        type Person struct {
+            Name       string
+            Street     string
+            City       string
+            PostalCode string
+        }
+
+        type APIPerson struct {
+            Name    string
+            Address *APIAddress
+        }
+
+        type APIAddress struct {
+            Street     string
+            City       string
+            PostalCode int
+        }
+wrapErrors: true
+success: |
+    // Code generated by github.com/jmattheis/goverter, DO NOT EDIT.
+
+    package generated
+
+    import (
+    	"fmt"
+    	execution "github.com/jmattheis/goverter/execution"
+    	"strconv"
+    )
+
+    type ConverterImpl struct{}
+
+    func (c *ConverterImpl) Convert(source []execution.Person) ([]execution.APIPerson, error) {
+    	structsAPIPersonList := make([]execution.APIPerson, len(source))
+    	for i := 0; i < len(source); i++ {
+    		structsAPIPerson, err := c.ConvertPerson(source[i])
+    		if err != nil {
+    			var errValue []execution.APIPerson
+    			return errValue, fmt.Errorf("error setting index %d: %w", i, err)
```

Comment:
With this, we could get an error like `error setting index 1: error setting field PostalCode: invalid number "abc"`. What do you think of a custom error type like this:

Then we could have a cooler error like error setting `error setting source[1].PostalCode: invalid number: "abc"` What do you think about that?

(untested, we probably have to reverse the path list, or only add the path in WithPath to the start of the list)

```go
func wrapPathError(path string, err error) error {
	var pathError *PathError
	if errors.As(err, &pathError) {
		return pathError.WithPath(path)
	}
	return &PathError{Err: err, Path: []string{path}}
}

type PathError struct {
	Err  error
	Path []string
}

func (e *PathError) WithPath(path string) *PathError {
	return &PathError{Err: e.Err, Path: append(e.Path, path)}
}

func (e *PathError) StringPath() string {
	return strings.Join(e.Path, ".")
}

func (e *PathError) Error() string {
	return fmt.Sprintf("PathError %s: %s", e.StringPath(), e.Err)
}

func (e *PathError) Unwrap() error {
	return e.Err
}
```

Reply:
> Then we could have a cooler error like error setting error setting source[1].PostalCode: invalid number: "abc" What do you think about that?

I like this idea, but it requires tons of extra coding and special error to be declared per generated code (or stashed in some library). We can do it as a separate investment ... for now I feel like it does not add value to core problem: troubleshooting errors.

Reply:
let's have this idea filed as a separate Issue, there might be some design/tuning needed here:
* generate code for wrapPathError and PathError struct in the converter's package ONCE (note: there might be multiple converters in the same package so that part requires cross-converter sync). This could also be stashed into a special goverter package - but that would require even more work (first public runtime package out of goverter repo, etc..).
* invoke the wrapPathError on fmt.Errorf directly.